rts/motoko-rts/src/gc/copying.rs

```diff
@@ -63,96 +141,28 @@ unsafe fn evac(
 
     // Update evacuated field
     *ptr_loc = skew(obj_loc);
-
-    // Update end of to-space
-    *end_to_space += obj_size_bytes.0 as usize
 }
 
-unsafe fn scav(
-    begin_from_space: usize,
-    begin_to_space: usize,
-    end_to_space: &mut usize,
-    obj: usize,
-) {
+unsafe fn scav<M: Memory>(mem: &mut M, begin_from_space: usize, begin_to_space: usize, obj: usize) {
     let obj = obj as *mut Obj;
 
     crate::visitor::visit_pointer_fields(obj, begin_from_space, |field_addr| {
-        evac(
-            begin_from_space,
-            begin_to_space,
-            end_to_space,
-            field_addr as usize,
-        );
+        evac(mem, begin_from_space, begin_to_space, field_addr as usize);
```

Comment:
Now `end_to_space` is maintained as heap ptr in mem, right?

Reply:
Right.